src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -43,8 +44,37 @@ symbolt generate_java_generic_typet::operator()(
   const java_class_typet &class_definition =
     to_java_class_type(pointer_subtype);
 
-  const irep_idt generic_name =
-    build_generic_name(existing_generic_type, class_definition);
+  // Small auxiliary function, to print a single argument name.
+  std::function<std::string(reference_typet)> build_argument_name =
+    [&](const reference_typet &argument)
```

Comment:
Removing `&`, thank you for catching that. For pulling it out into a static function - there is another lambda function below it, I thought for the consistency it may make sense to keep it here. Or is it extensive enough to pull out?

Reply:
I don't have a strong preference. It is quite long but I also agree about the consistency - as you think best :)

Reply:
I agree it would be best to pull it in a static function: it's quite long for a lambda and the function it's in is getting too long. Regarding the other lambda below, I think it should go directly in the `for_each` argument since it is used only once.

Reply:
Pulling out as a static function.